sktime/forecasting/model_selection/_split.py

```diff
@@ -521,6 +522,35 @@ def get_fh(self) -> ForecastingHorizon:
     def __repr__(self) -> str:
         return _repr(self)
 
+    @staticmethod
+    def _get_train_window(
+        y: ACCEPTED_Y_TYPES, train_start: int, split_point: int
+    ) -> np.ndarray:
+        """Get train window.
+
+        For formal definition of the train window see docstring of the `BaseSplitter`
+
+        Parameters
+        ----------
+        y : pd.Index
```

Comment:
ok, thanks!

Now here is what confused me. The code looks like it's written for `pd.Index`.
But, the type hint says that `y` can be `ACCEPTED_Y_TYPES = Union[pd.Series, pd.DataFrame, np.ndarray, pd.Index]`.
How does this fit together, and with the docstring? Is it correct that we also want to be able to pass all this stuff, or do we really only want an index?

Reply:
I went through all functions/methods that use `y` and replaced type suggestions and corrected docstrings.

Reply:
ah, that makes more sense!

So if `y` is not an index, it gets the index from `y`?

Reply:
I have renamed `_check_y` to more clear `_get_index_of_time_series`. This function is used in every public method of splitters before going to any internal processing.

Reply:
I see. But *this* is a straightforward duplication then! Namely, of `datatypes.utilities.get_time_index`.

Reply:
this change is straightforward. replaced `_get_index_of_time_series` with `sktime.datatypes._utilities.get_index_for_series`

Reply:
ok, I think with me now better understanding what is going on (thanks for clarifying and docstrings!), and with the duplication removed, I think we are good now! Give me some time to think about it still, but I feel we're ready.